### DISCUSSION TITLE
Bombardment of 403, 503 and captcha errors

### DISCUSSION THREAD

**Discussion by Vuand** [jef/streetmerchant#1413] [General]
Recently deleted streetmerchant and reinstalled it with the updated version which added the 3060 ti, however, I'm being bombarded with 403, 503, and captcha errors that I never got before on sites such as best buy, newegg, and amazon. How do I fix this and why has this only become a problem with the newest update of streetmerchant?

![Screenshot (2)](https://user-images.githubusercontent.com/75465522/102138669-8e6ff480-3e11-11eb-9be2-b1adc98acdca.png)

**Comment by law93** [comment]
In a first place, go to : https://www.google.com/search?q=what+is+my+user+agent 
and copy past on your .env for USER_AGENT=""

**Comment by gigi2006** [comment]
> In a first place, go to : https://www.google.com/search?q=what+is+my+user+agent
> and copy past on your .env for USER_AGENT=""

he doesn't need that anymore since the latest update a few days ago. there a random user agent is taken.

he can USER_AGENT="" still use if he wants. (I do by the way also still) but as far as I have read, Jef will disable this function from 4.0.

**Comment by Vuand** [comment]
I think the random user agent is the culprit for these errors. I reverted back to version 3,1 and there were no errors but as soon as I added the random user agent that you guys suggest on the previous version of street merchant, I kept getting these captchas, 403, and 503 errors. Why is this the case? Is there a different solution for the latest version of street merchant because having a random user agent is not working

**Comment by MidnightsFX** [comment]
Newegg, and other merchants will eventually start blocking your IP for at least a short period of time, which can result in the 403 or 503 errors.

You can avoid hitting those IP bans by going through a handful or proxies etc or tuning your checking frequency to be slower.

**Comment by smazurov** [comment]
@Vuand as others have stated in this thread,  add this to you `.env`, on a new line, at the bottom: 
```
USER_AGENT=""
```
To a valid user agent (which you can google for)

**Comment by rattlion** [comment]
In my experience, and with Newegg in particular, UA has not helped.

**Comment by jef** [comment]
I've recently updated the random UAs to use the top 50 browsers. This should help with this.

You may still get issues, but definitely should help.
